crates/github-actions-expressions/src/lib.rs

```diff
@@ -674,6 +674,346 @@ impl From<bool> for Expr<'_> {
     }
 }
 
+/// The result of evaluating a GitHub Actions expression.
+///
+/// This type represents the possible values that can result from evaluating
+/// constant-reducible GitHub Actions expressions.
+#[derive(Debug, Clone, PartialEq)]
+pub enum EvaluationResult {
+    /// A string value (includes both string literals and stringified other types)
+    String(String),
+    /// A numeric value
+    Number(f64),
+    /// A boolean value
+    Boolean(bool),
+    /// The null value
+    Null,
+}
+
+impl EvaluationResult {
+    /// Convert the evaluation result to GitHub Actions string representation.
+    ///
+    /// This follows GitHub Actions' string conversion rules:
+    /// - Strings are returned as-is (without quotes)
+    /// - Numbers are converted to string representation
+    /// - Booleans become "true" or "false"
+    /// - Null becomes an empty string
+    pub fn to_github_string(&self) -> String {
+        match self {
+            EvaluationResult::String(s) => s.clone(),
+            EvaluationResult::Number(n) => {
+                // Format numbers like GitHub Actions does
+                if n.fract() == 0.0 {
+                    format!("{}", *n as i64)
+                } else {
+                    n.to_string()
+                }
+            }
+            EvaluationResult::Boolean(b) => b.to_string(),
+            EvaluationResult::Null => String::new(),
+        }
+    }
+
+    /// Convert to a boolean following GitHub Actions truthiness rules.
+    ///
+    /// GitHub Actions truthiness:
+    /// - false and null are falsy
+    /// - Numbers: 0 is falsy, everything else is truthy
+    /// - Strings: empty string is falsy, everything else is truthy
+    pub fn to_boolean(&self) -> bool {
```

Comment:
Nitpick: rename to `as_boolean` since `to_*` usually implies ownership transfer in Rust:

```suggestion
    pub fn as_boolean(&self) -> bool {
```

Reply:
Fixed in https://github.com/zizmorcore/zizmor/pull/1088/commits/f04a001888fed50bcd80b9600806f54d072a47fd.